### ISSUE TITLE
Parsing Lao words

### ISSUE THREAD

**Issue by bountonw** [bountonw/translate-tooling#61]
@mattleff 

Currently none of our Lao files are sharable. 

We need to begin the process of transforming the Lao text into something sharable. When copying text from the Lao Bible, it automatically inserts a `U+200B` _Zero Width Space_ between words. This allows for typesetting engines to be able to line wrap.

**Example:** `“ເພາະ​ພວກເຂົາ​ໄດ້​ເອົາ​ອົງພຣະຜູ້ເປັນເຈົ້າ​ຂອງ​ຂ້ານ້ອຍ​ໄປ​ແລ້ວ ແລະ ຂ້ານ້ອຍ​ບໍ່​ຮູ້ຈັກ​ວ່າ​ພວກເຂົາ​ເອົາ​ພຣະອົງ​ໄປ​ໄວ້​ຢູ່​ໃສ”.` There are 18 Zero Width Spaces in this short section. 

How do we go about cleaning up the files? Do we have clean `.md` files without the ZWS and then insert as an intermediary step for outputting into the various formats? Or do we have ZWS in the `.md` files and ensuring that they are entered properly then becomes part of the file updating process?

**Comment by mattleff** [comment]
@bountonw That's quite interesting. I wonder if we should actually consider the ZWS approach for Thai text as well. If text layout engines (InDesign, Latex, or any other) were able to use the ZWS's to break lines as appropriate that could give us a lot more flexibility in rendering options and the consistency of capturing line-break positions in version control. We could potentially backfill ZWS's into the Thai text using our current word break engines (Swath, etc.) and then fix from there. The ZWS's would also become word boundary markers which would significantly ease dictionary-based spell-checking. 

What would be the downsides of having ZWS's stored in the text everywhere?

**Comment by bountonw** [comment]
I don’t think it is standard to do for Thai and my EAM only allows me to worry about one problem at a time. 😀 For Lao, however, it seems to be more common as I have even seen it on other Lao websites. Not sure how they enter it. I scraped the Lao Bible once and I think I have the files still. So that could be a source of words. Do you want me to send it to you? There is a new version of the Lao Bible online as well. So that could possibly provide an additional word list. My data is raw so you might do better just scraping from scratch.

**Comment by bountonw** [comment]
@mattleff I see my RAM was overwhelmed in my last comment. It has been edited. 

If we did have ZWS in Thai it would help in collaboration via google docs. As it is google docs splits the Thai words in random places. But if it is a separate process for Thai and Lao, the Lao is more urgent.

**Comment by mattleff** [comment]
Talked about this and concluded:
1. Don't store ZWS in text. Add lint rule against ZWS.
2. Have script insert ZWS using a Lao word segmenter (maybe https://codeberg.org/mekong-lang/chamkho?) and a custom Lao dictionary.
3. Future: If there are cases in Lao that require overriding ZWS insertion, use pipe (`|`) as separator token to allow manually overriding ZWS insertion.
